client.go

```diff
@@ -2066,6 +2079,10 @@ func (c *Client) executeRequestMiddlewares(req *Request) (err error) {
 // Executes method executes the given `Request` object and returns
 // response or error.
 func (c *Client) execute(req *Request) (*Response, error) {
+	if err := c.circuitBreaker.allow(); err != nil {
```

Comment:
@segevda Please add a `nil` check. By default, the circuit breaker is not enabled.

Reply:
@jeevatkm I know it's not very idiomatic, but the `nil` check is inside the function. I chose this path to not add too much cognitive load to `execute` with a small feature. If you think it's better to add here I will.

Reply:
@segevda okay, let's keep it as is.